knox/auth.py

```diff
@@ -56,15 +56,24 @@ def authenticate_credentials(self, token):
         '''
         msg = _('Invalid token.')
         token = token.decode("utf-8")
+
+        try:
+            digest = hash_token(token)
+        except (TypeError, binascii.Error):
+            raise exceptions.AuthenticationFailed(msg)
+
+        # Migrate tokens with this digest that were created prior to 3a1bc58
+        for auth_token in get_token_model().objects.filter(
+            digest=digest, token_key=token[:8]
```

Comment:
I'm using a direct filter of the digest here. Not exactly sure why that is not done in the filter below - it can be done directly now that a salt per token is not used. Any thoughts @johnraz ?

Reply:
In fact, now that a salt is not used I don't think that storing the `token_key` is necessary at all.